fla/models/utils.py

```
@@ -2,149 +2,303 @@
+                )
+                self.states.append(state)
+            else:
+                # update the number of seen tokens
+                if layer_idx == len(self.states) - 1:
+                    self._seen_tokens += offset
+                state = self.states[layer_idx]
+                if recurrent_state is not None:
+                    state['recurrent_state'] = recurrent_state
+                if attn_state is not None:
+                    if window_size is not None and state['attn_state'][0].shape[1] == window_size:
+                        for i, (old_state, new_state) in enumerate(zip(state['attn_state'], attn_state)):
```

Ruff (0.12.2)

`zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)